lib/Serialization/DeserializeSIL.cpp

```diff
@@ -430,9 +430,7 @@ SILDeserializer::readLoc(unsigned kind, SmallVectorImpl<uint64_t> &scratch) {
   case SILLocation::RegularKind:
     return RegularLocation(FNameLoc, Implicit);
   }
-
-  // LocationKind was not a recognized SILLocation::LocationKind.
-  return std::nullopt;
+  llvm_unreachable("Invalid LocationKind");
```

Comment:
We should not use llvm_unreachable to deal with invalid input. Code like this can crash long-running tools like LLDB which makes for a very unpleasant user experience. Can you just return an error instead?